Content.Server/GameObjects/Components/Power/NewPower/BatteryDischargerComponent.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.NewPower
+{
+    /// <summary>
+    ///     Pulls charge from a <see cref="BatteryComponent"/> to supply power with a <see cref="PowerSupplierComponent"/>.
+    /// </summary>
+    [RegisterComponent]
+    public class BatteryDischargerComponent : Component
+    {
+        public override string Name => "BatteryDischarger";
+
+        /// <summary>
+        ///     The battery that this is trying to take power from.
+        /// </summary>
+        [ViewVariables]
+        private BatteryComponent _battery;
+
+        /// <summary>
+        ///     The supplier power is released with.
+        /// </summary>
+        [ViewVariables]
+        private PowerSupplierComponent _supplier;
+
+        /// <summary>
+        ///     The amount of power discharged by the supplier when the battery has power.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int EnabledSupplyRate { get => _enabledSupplyRate; set => SetEnabledSupplyRate(value); }
+        private int _enabledSupplyRate;
+
+        /// <summary>
+        ///     Whether or not the battery has charge to allow the supplier to give
+        ///     power to its powernet.
+        /// </summary>
+        [ViewVariables]
+        private bool _hasCharge;
```

Comment:
Is there a reason for this to be its own var, instead of just checking `_battery.BatteryState != BatteryState.Empty`?